<launch>

  <!-- Controller parameters -->
  <!-- <arg name="controller"	default="aist_ftsensor_controller"/> -->
  <arg name="controller"	default="force_torque_sensor_controller"/>
  <arg name="publish_rate"	default="125"/>
  <arg name="calib_file"	default=""/>
  <arg name="box_filter_size"	default="1"/>

  <!-- Driver parameters -->
  <arg name="frame_id"		default="ftsensor_wrench_link"/>
  <arg name="dev"		default="/dev/ttyUSB0"/>
  <arg name="baud"		default="921600"/>
  <arg name="reset_offsets"	default="false"/>
  <arg name="avg_npoints"	default="1"/>

  <node name="dynpick_hardware_interface"
        pkg="aist_ftsensor" type="dynpick_driver" output="screen">
    <param name="frame_id"	value="$(arg frame_id)"/>
    <param name="rate"		value="$(arg publish_rate)"/>
    <param name="dev"		value="$(arg dev)"/>
    <param name="baud"		value="$(arg baud)"/>
    <param name="reset_offsets"	value="$(arg reset_offsets)"/>
    <param name="avg_npoints"	value="$(arg avg_npoints)"/>
  </node>

  <rosparam if="$(eval calib_file != '')"
	    file="$(env HOME)/.ros/aist_ftsensor/$(arg calib_file).yaml"/>

  <rosparam file="$(dirname)/../config/ftsensor_controller.yaml"
  	    subst_value="true"/>
  <node name="controller_spawner"
        pkg="controller_manager" type="spawner" output="screen"
  	args="ftsensor_controller"/>
</launch>
